src/utils.ts

```diff
@@ -0,0 +1,111 @@
+    if (tempDirs[sharedKey]) {
+        return Promise.resolve(tempDirs[sharedKey]);
+    }
+
+    return new Promise((resolve, reject) => {
+        temp.track();
+        temp.mkdir(sharedKey, (err, dirPath) => {
+            if (err) {
+                reject(err);
+            }
+            else {
+                tempDirs[sharedKey] = dirPath;
+                resolve(dirPath);
+            }
+        })
+    });
+}
+
+export function openAndRead(path: string, offset: number, length: number, position: number): Promise<Buffer> {
+    return new Promise((resolve, reject) => {
+        fs.open(path, 'r', (err, fd) => {
+            if (err) {
+                reject(err);
+                return;
+            }
+
+            const buffer = new Buffer(length);
+            fs.read(fd, buffer, offset, length, position, (err, bytesRead, buffer) => {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+
+                resolve(buffer);
+            });
+        });
+    })
+}
+
+export async function verifyFileHeader(filePath: string, expectedHeader: Buffer | number[], offset?: number): Promise<boolean> {
+    const bufferExpectedHeader = isArray(expectedHeader) ? new Buffer(<number[]>expectedHeader) : <Buffer>expectedHeader;
+    const header = await openAndRead(filePath, 0, bufferExpectedHeader.length, offset);
+    return header.compare(bufferExpectedHeader) == 0;
+}
+
+export class StatusBarProgressionMessage {
+    private _statusBarItem: vscode.StatusBarItem;
+
+    constructor(initialMessage?: string) {
+        this._statusBarItem = vscode.window.createStatusBarItem();
+        if(initialMessage) {
+            this._statusBarItem.text = initialMessage;
+            this._statusBarItem.show();
+        }
+    }
+
+    /**
+     * Updates the displayed message.
+     * @param newMessage The new message to display
+     */
+    public update(newMessage: string) {
+        if(!this._statusBarItem) {
+            return;
+        }
+
+        this._statusBarItem.text = newMessage;
+        this._statusBarItem.show();
+    }
+
+    /**
+     * Marks the progression as being finished. If a message is specified, it is
+     * shown temporarily before the item disappears.
+     * 
+     * Note that a message should always be provided if an external message
+     * indicating a failure won't be presented to the user.
+     * @param finalMessage The last message to show for a short period
+     * @param delay The amount of time the final message should be shown
+     */
+    public finish(finalMessage?: string, delay: number = 5000) {
+        if(!this._statusBarItem) {
+            return;
+        }
+
+        if(finalMessage) {
+            this.update(finalMessage);
+            setTimeout(() => this._dispose(), delay);
+        }
+        else {
+            this._dispose();
+        }
+    }
+
+    private _dispose() {
```

Comment:
style: don't really need the leading underscore in typescript since we have the access modifier to tell us this is private.